database/warehouse/events/unit_died_event.py

```diff
@@ -38,6 +36,59 @@ class unit_died_event(Base):
     info_id = Column(Integer, ForeignKey("replay.info.primary_id"))
     info = relationship("info", back_populates="unit_died_events")
 
-    player_id = Column(Integer, ForeignKey("replay.player.primary_id"))
-    killing_player = relationship("player", back_populates="unit_died_events")
+    @classmethod
+    @property
+    def __tableschema__(self):
+        return "events"
+
+    @classmethod
+    async def process(cls, replay, session):
+        events = replay.events_dictionary['UnitDiedEvent']
+
+        _events = []
+        for event in events:
+            data = cls.get_data(event)
+            parents = await cls.process_dependancies(event, replay, session)
+
+            _events.append(cls(**data, **parents))
+
+        session.add_all(_events)
+
+    @classmethod
+    async def process_dependancies(cls, event, replay, session):
+        _info, _unit, _killing_unit = replay.filehash, event.unit_id, event.killing_unit_id
+        parents = defaultdict(lambda:None)
+
+        info_statement = select(info).where(info.filehash == _info)
+        info_result = await session.execute(info_statement)
+        _info = info_result.scalar()
+        parents['info_id'] = _info.primary_id
+
+        unit_statement = select(object).where(
+                and_(object.info_id == _info.primary_id, object.id == _unit))
+        unit_result = await session.execute(unit_statement)
+        _unit = unit_result.scalar()
+        parents['unit_id'] = _unit.primary_id
+
+        # Not all units have a killer.
+        if not _killing_unit:
```

Comment:
In case of 'suicide' death event, ignore killing_unit. Perhaps self reference in this case?